### PR TITLE
Tests#90 renew references

### PR DESCRIPTION
**Pull request by Gymbotan**
New KeyDerivationLib submodule. 
Rename parameters according to new classes names from KeyDerivationLib. 
Change Enumerator to KeyCreationReason. 
Change interface method DeriveKeyPair. Add parameter int keyIndex.
Edit tests according to new changes.

### CODE REVIEW

File: TuviPgpLib/Entities/KeyCreationReason.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+///////////////////////////////////////////////////////////////////////////////
+
+namespace TuviPgpLib.Entities
+{
+    public enum KeyCreationReason : int
```
Comment (anatolse) [inline TuviPgpLib/Entities/KeyCreationReason.cs:19]: I thought we move this to the client size. The question: how to extend this enum for user-specific application needs?

File: TuviPgpLibImpl/EccPgpContext.cs [hunk 2]
```diff
@@ -151,21 +192,51 @@ private PgpSignatureSubpacketGenerator CreateSignatureSubpacketGenerator(long ex
...
             }
 
-            return subpacketGenerator;
+            foreach (PgpSecretKey item in EnumerateSecretKeys(signer))
+            {
+                if (item.IsSigningKey && !item.IsMasterKey)
+                {
+                    PgpPublicKey publicKey = item.PublicKey;
+                    if (!publicKey.IsRevoked() && !OpenPgpContext.IsExpired(publicKey))
+                    {
```
Comment (ValeraFinebits) [inline TuviPgpLibImpl/EccPgpContext.cs:233]: Is it possible to get rid of duplicate code here?

File: TuviPgpLibImpl/EccPgpContext.cs [hunk 3]
```diff
@@ -151,21 +192,51 @@ private PgpSignatureSubpacketGenerator CreateSignatureSubpacketGenerator(long ex
...
-            subpacketGenerator.SetPreferredHashAlgorithms(false, this.EnabledDigestAlgorithms.Select(e => (int)e).ToArray());
+            foreach (PgpSecretKeyRing item in EnumerateSecretKeyRings(mailbox))
+            {
+                foreach (PgpSecretKey secretKey in item.GetSecretKeys())
+                {
+                    if (secretKey.IsSigningKey && !secretKey.IsMasterKey)
+                    {
+                        PgpPublicKey publicKey = secretKey.PublicKey;
+                        if (!publicKey.IsRevoked() && !OpenPgpContext.IsExpired(publicKey))
+                        {
```
Comment (ValeraFinebits) [inline TuviPgpLibImpl/EccPgpContext.cs:210]: Is it possible to get rid of duplicate code here?

Comment (Gymbotan) [reply TuviPgpLibImpl/EccPgpContext.cs]: Create IsSigningNotMaster() method

**Review by ValeraFinebits — APPROVED** [review]

**Review by anatolse — APPROVED** [review]

**Pull request merged**